AppCenterDataStorage/AppCenterDataStorage/MSDataStore.m

```diff
@@ -658,4 +664,117 @@ - (void)authTokenContext:(MSAuthTokenContext *)__unused authTokenContext didUpda
+
+                                                    } else {
+                                                      [self.dataOperationProxy.documentStore deleteWithToken:tokenResponses.tokens[0]
+                                                                                                  documentId:operation.documentId];
+                                                    }
+
+                                                  } else if (documentWrapper.error.errorCode == MSACDocumentNotFoundErrorCode ||
+                                                             documentWrapper.error.errorCode == MSACDocumentConflictErrorCode) {
+                                                    MSLogError(
+                                                        [MSDataStore logTag],
+                                                        @"Failed to call Cosmos delete API. Remote operation failed with error code: %ld",
+                                                        documentWrapper.error.errorCode);
+                                                    [self.dataOperationProxy.documentStore deleteWithToken:tokenResponses.tokens[0]
+                                                                                                documentId:operation.documentId];
+                                                  } else {
+                                                    MSLogError([MSDataStore logTag], @"Failed to call Cosmos create or replace API: %@",
+                                                               [documentWrapper.error.error localizedDescription]);
+                                                  }
+                                                }];
+                                       } else if ([operation.operation isEqualToString:kMSPendingOperationDelete]) {
+
+                                         // Perform delete operation
+                                         [MSDataStore
+                                             deleteWithPartition:(NSString *)operation.partition
+                                                      documentId:(NSString *)operation.documentId
+                                                    writeOptions:(MSWriteOptions * _Nullable) writeOptions
+                                               completionHandler:(MSDocumentWrapperCompletionHandler) ^ (MSDocumentWrapper *
+                                                                                                         documentWrapper) {
+                                                 if (!documentWrapper.error) {
+                                                   [self.dataOperationProxy.documentStore deleteWithToken:tokenResponses.tokens[0]
+                                                                                               documentId:operation.documentId];
+                                                 } else if (documentWrapper.error.errorCode == MSACDocumentNotFoundErrorCode ||
```

Comment:
This block is the same as the one on line 733.  Perhaps consider refactoring it into a different method?